cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -473,6 +472,12 @@ public void callArtifactPluginMethod(HttpRequest request, HttpResponder responde
                                       "Received empty request body.");
     }
 
+    // should not happen
+    if (!pluginService.isRunning()) {
```

Comment:
if we're worried about this, the logic should be in PluginService rather than in the caller